eo-maven-plugin/src/main/java/org/eolang/maven/HashOfTag.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.jcabi.log.Logger;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Objects;
+import java.util.Scanner;
+
+/**
+ * Hash of tag.
+ * @since 0.26
+ */
+public class HashOfTag {
+    /**
+     * Hash.
+     */
+    private final String hash;
+
+    /**
+     * Tag.
+     */
+    private final String tag;
+
+    /**
+     * Constructor.
+     * @param tag Tag
+     * @throws IOException if fails
+     * @throws TagNotFoundException if fails
+     * @checkstyle NonStaticMethodCheck (20 lines)
+     */
+    public HashOfTag(final String tag) throws IOException, TagNotFoundException {
+        this.tag = tag;
+        this.hash = this.getSha();
+    }
+
+    /**
+     * Accessor.
+     * @return Template
+     */
+    public String getHash() {
+        return this.hash;
+    }
+
+    /**
+     * Hash of tag.
+     * @return SHA of commit
+     * @throws IOException if fails
+     * @throws TagNotFoundException if fails
+     * @checkstyle NonStaticMethodCheck (20 lines)
```

Comment:
@yasamprom here Checkstyle is giving you a hint that your design is not good. This class should look like this: 

```

```

Please, read blog posts related to object immutability and static methods: https://www.yegor256.com/tag/oop.html